### ISSUE TITLE
make auth system

### ISSUE THREAD

**Issue by nerdylive123** [fencepos/fencepos#17]

**Comment by nerdylive123** [comment]
- New TODO: add tests too

**Comment by m-gschwandtner** [comment]
hi, passport is a pretty good option if you want to add stuff like OAuth, or different auth methods in general,
depending on your use case, of course

https://www.npmjs.com/package/passport
https://www.passportjs.org/packages/

best of luck

**Comment by jxpsert** [comment]
yep, we've been switching between stuff for a while. if you're an expert in any of this, feel free to contribute 😅

**Comment by m-gschwandtner** [comment]
sadly not a lot of time currently, but I wanted to build a bookmark manager a while ago,
has functioning local auth + GitHub auth (/server/utils/passport.ts), docker setup and prisma

removed the git folder and zipped it since i commited some api key ...

uses typescript and svelte, but most of the passport, docker and prisma config should work fine for your use case,
feel free to use anything from the code

[zen-bookmarks.zip](https://github.com/fencepos/fencepos/files/10232203/zen-bookmarks.zip)

**Comment by nerdylive123** [comment]
thanks we'll look at it
